thoth/common/openshift.py

```diff
@@ -526,6 +540,43 @@ def _transform_cronjob_to_job(item: Dict[str, Any]) -> Dict[str, Any]:
         job_spec["metadata"]["generateName"] = item["metadata"]["name"] + "-"
         return job_spec
 
+    def _submit_workflow(
+        self,
+        wf: argo.models.V1alpha1Workflow,
+        namespace: Optional[str] = None,
+        parameters: Optional[Dict[str, str]] = None,
+    ):
+        """Submit an Argo Workflow."""
+        if namespace is None:
+            if not self.frontend_namespace:
+                raise ConfigurationError(
+                    "No frontend namespace configured to run graph-refresh job"
+                )
+            namespace = self.frontend_namespace
+
+        parameters = parameters or {}
+
+        new_parameters: List[argo.models.V1alpha1Parameter] = []
+        for name, value in parameters.items():
+            param = argo.models.V1alpha1Parameter(
+                name=name,
+                value=value
+            )
+            new_parameters.append(param)
+
+        for p in wf.spec.arguments.parameters:
+            if p.name in parameters:
+                continue  # overridden
+            elif not p.value and not p.default:
+                raise Exception("Required parameter {p.name}")
```

Comment:
f-string needed. Probably a separate exception type would be appropriate for this.

Reply:
True that! Thanks :+1: